src/diff.c

```diff
@@ -543,6 +543,15 @@ diff_read(struct view *view, struct buffer *buf, bool force_stop)
 			}
 		}
 
+		if (view->env->blame_lineno) {
```

Comment:
Just curious, why does the code go here? Is it just stylistic?

Also, `state->pos` is really bizarre :)

Reply:
Agreed but it's almost as old as Tig. We need to run this code after the file has been read in order to know how many lines there are, so we have to be in the `if (!buf) {`.